pkg/route/ingress/ingress.go

```diff
@@ -545,25 +555,49 @@ func newRouteForIngress(
 	}
 
 	t := true
-	return &routev1.Route{
-		ObjectMeta: metav1.ObjectMeta{
-			GenerateName: ingress.Name + "-",
-			Namespace:    ingress.Namespace,
-			Labels:       ingress.Labels,
-			Annotations:  ingress.Annotations,
-			OwnerReferences: []metav1.OwnerReference{
-				{APIVersion: "networking.k8s.io/v1beta1", Kind: "Ingress", Controller: &t, Name: ingress.Name, UID: ingress.UID},
+	if hostIsWildcard {
+		return &routev1.Route{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: ingress.Name + "-",
+				Namespace:    ingress.Namespace,
+				Labels:       ingress.Labels,
+				Annotations:  ingress.Annotations,
+				OwnerReferences: []metav1.OwnerReference{
+					{APIVersion: "networking.k8s.io/v1beta1", Kind: "Ingress", Controller: &t, Name: ingress.Name, UID: ingress.UID},
+				},
 			},
-		},
-		Spec: routev1.RouteSpec{
-			Host: rule.Host,
-			Path: path.Path,
-			To: routev1.RouteTargetReference{
-				Name: path.Backend.ServiceName,
+			Spec: routev1.RouteSpec{
+				Host: host,
+				Path: path.Path,
+				To: routev1.RouteTargetReference{
+					Name: path.Backend.ServiceName,
+				},
+				Port:           port,
+				TLS:            tlsConfigForIngress(ingress, rule, tlsSecret),
+				WildcardPolicy: routev1.WildcardPolicySubdomain,
 			},
-			Port: port,
-			TLS:  tlsConfigForIngress(ingress, rule, tlsSecret),
-		},
+		}
+	} else {
+		return &routev1.Route{
+			ObjectMeta: metav1.ObjectMeta{
```

Comment:
The route resource returned on line 581 only differs in spec from the route resource returned on line 559, right? Probably best to use an if statement on `hostIsWildcard` to conditionally create the route's spec so that only one return statement is needed (rather than 2 return statements with lots of common code).